core/src/main/scala/quasar/fs/InMemory.scala

```
@@ -213,10 +214,18 @@ object InMemory {
         import quasar.LogicalPlan._
         import quasar.std.StdLib.set.{Drop, Take}
         import quasar.std.StdLib.identity.Squash
+        // Side-step https://issues.scala-lang.org/browse/SI-9581 by avoiding values of limit
```

I think this is the right place to address this issue. However your comment made me realize that it would be more correct to return a `FileSystemError` instead of silently using a smaller value.